plugin.go

```diff
@@ -2,39 +2,58 @@
 // also, the interfaces for input and output plugins.
 package plugin
 
+import "C"
```

Comment:
This is coming in from `C.size_t` use here:
```
func (p *pluginInstance) callback(data *unsafe.Pointer, csize *C.size_t) int {
```

If there's some reason we want to contain CGO to cshared.go, that's doable.